File: leetcode/1133-largest-unique-number/solution.cpp
int largestUniqueNumber(vector<int> nums) {
    vector<int> freq(1001, 0);
    for (int num : nums) {
        freq[num]++;
    }

    int ans = -1;
    for (int num : nums) {
        if (freq[num] == 1) {
            ans = max(ans, num);
        }
    }

    return ans;
}

File: leetcode/1133-largest-unique-number/solution.py
def largestUniqueNumber(self, nums: List[int]) -> int:
    counts = [0] * 1001

    for num in nums:
        print(num)
        counts[num] += 1

    ans = -1
    for num in range(0, 1001):
        if counts[num] == 1:
            ans = max(ans, num)

    return ans
